src/js/msp.js

```diff
@@ -304,14 +307,14 @@ const MSP = {
         return bufferOut;
     },
     send_message(code, data, callback_sent, callback_msp, doCallbackOnError) {
-        if (code === undefined || !serial.connectionId || CONFIGURATOR.virtualMode) {
+        const connected = import.meta.env ? serial.connected : serial.connectionId;
```

Comment:
```suggestion
        const connected = import.meta.env ? serial.connected : !!serial.connectionId;
```

Reply:
Looking at the code now, it should probably be 
```suggestion
        const connected = import.meta.env ? serial.connected : !serial.connectionId;
```

Reply:
Hmm, actually, is should probably be `!`
```suggestion
        const connected = import.meta.env ? serial.connected : !serial.connectionId;
```